src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSSynchronizationContext.cs

```diff
@@ -215,7 +216,10 @@ public override void Send(SendOrPostCallback d, object? state)
                     Environment.FailFast($"JSSynchronizationContext.Send failed, ManagedThreadId: {Environment.CurrentManagedThreadId}. {Environment.NewLine} {Environment.StackTrace}");
                 }
 
+                var threadFlag = Monitor.ThrowOnBlockingWaitOnJSInteropThread;
```

Comment:
I'm re-thinking if this should be allowed after all. Because that could call code on the other thread, which would be long blocking.

At the moment, this is necessary for the HTTP/WS clients to work across threads. But now I will change the implementation to use emscripten instead of `JSSynchronizationContext` to make that dispatch.
It will also have the same problem, tho.

So I guess we merge this for now and I will improve it bit later.